src/dstack/_internal/core/models/configurations.py

```diff
@@ -162,6 +164,58 @@ class RateLimit(CoreModel):
     ] = 0
 
 
+class ProbeConfig(CoreModel):
+    type: Literal["http"]  # expect other probe types in the future, namely `exec`
+    url: Annotated[str, Field(description="The URL to request")] = "/"
+    timeout: Annotated[
+        Union[int, str],
+        Field(description=("Maximum amount of time the HTTP request is allowed to take")),
+    ] = "10s"
```

Comment:
Please avoid putting defaults directly into models. We did it in the past and it caused lots of issues. Primarily being unable to update defaults on the server-side and detecting if the user set the field or it was set by model defaults. Fields should be Optional[] with defaults set on the server.